rebar.config

```diff
@@ -157,6 +157,62 @@
     [ {provider_hooks,
        [ {post, [ {compile, {pc, compile}},
                   {clean, {pc, clean}} ] }] } ]},
+
+   %% Compile with port_compiler instead of Makefile
+   {override, snappy,
+    [ {pre_hooks, []},
+      {post_hooks, []},
+      {plugins, [pc]},
+      {artifacts, ["priv/snappy_nif.so"]},
+      {port_specs, [{".*",
+                     "priv/snappy_nif.so",
+                     ["c_src/*.cc"],
+                     []
+                    }]},
+      {provider_hooks,
+       [ {post, [ {compile, {pc, compile}},
+                  {clean, {pc, clean}} ] }] } ]},
+
+   %% Compile with port_compiler instead of Makefile
+   {override, lz4,
+    [ {pre_hooks, []},
+      {post_hooks, []},
+      {plugins, [pc]},
+      {artifacts, ["priv/lz4_nif.so"]},
+      {port_specs, [{".*",
+                     "priv/lz4_nif.so",
+                     ["c_src/*.c"],
+                     []
+                    }]},
+      {provider_hooks,
+       [ {post, [ {compile, {pc, compile}},
+                  {clean, {pc, clean}} ] }] } ]},
+
+   %% Compile with port_compiler instead of Makefile
+   {override, exml,
```

Comment:
Could it be done in exml directly instead of the override here?

Reply:
yep, this PR is more like a prototype.
same with fast_tls part, we should just make PR.

But I am not sure if lz4 and snappy maintainers are happy with pc...

Reply:
> But I am not sure if lz4 and snappy maintainers are happy with pc...

Did you ask them?

Reply:
nope. I can make PRs to all the libs and check...

Reply:
Please do :)

Reply:
BTW if there's interest from ESL to be added as maintainers for snappy just drop me an email.